### PR TITLE
add option to hide /debug/ pages

### LINKED ISSUE #5305: [feature request] Add configuration option to disable the `/debug/vars` HTTP API endpoint
Currently the `/debug/vars` endpoint is not protected by HTTP authentication, which could expose inadvertent debug information to anyone with network access to the InfluxDB instance. Adding the ability to disable this endpoint via configuration would prevent this issue.

### PR DESCRIPTION
**Pull request by csko**
closes #5305

###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

###### Required only if applicable
- [x] Config changes: update sample config (`etc/config.sample.toml`), server `NewDemoConfig` method, and `Diagnostics` methods reporting config settings, if necessary
- [ ] [InfluxData Documentation](https://github.com/influxdata/docs.influxdata.com): issue filed or pull request submitted

### CODE REVIEW

File: etc/config.sample.toml [hunk 1]
```diff
@@ -219,6 +219,10 @@
   # troubleshooting and monitoring.
   # pprof-enabled = true
 
+  # Determines whether the debug endpoint is enabled.  This endpoint is used for
+  # gathering runtime statistics.
+  # pprof-enabled = true
```
Comment (jsternberg) [inline etc/config.sample.toml]: This should be updated to `debug-enabled` rather than `pprof-enabled`.

File: services/httpd/handler.go [hunk 2]
```diff
@@ -256,9 +256,9 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	if strings.HasPrefix(r.URL.Path, "/debug/pprof") && h.Config.PprofEnabled {
 		h.handleProfiles(w, r)
-	} else if strings.HasPrefix(r.URL.Path, "/debug/vars") {
+	} else if strings.HasPrefix(r.URL.Path, "/debug/vars") && h.Config.DebugEnabled {
```
Comment (jsternberg) [inline services/httpd/handler.go]: Should setting `debug-enabled` to false also disable `/debug/pprof`?

File: services/httpd/config.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ type Config struct {
 	LogEnabled         bool   `toml:"log-enabled"`
 	WriteTracing       bool   `toml:"write-tracing"`
 	PprofEnabled       bool   `toml:"pprof-enabled"`
+	DebugEnabled       bool   `toml:"debug-enabled"`
```
Comment (jsternberg) [inline services/httpd/config.go]: The default for this should be set to `true`.

File: services/httpd/handler.go (hunk 2 above)
Comment (jwilder) [inline services/httpd/handler.go]: I think that is what `pprof-enabled` does.

**Comment by jwilder** [comment]
Closing due to inactivity.

**Pull request closed without merging**